app/src/main/java/org/oppia/app/player/audio/AudioViewModel.kt

```diff
@@ -1,17 +1,118 @@
 package org.oppia.app.player.audio
 
 import androidx.databinding.ObservableField
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.Transformations
 import androidx.lifecycle.ViewModel
 import org.oppia.app.fragment.FragmentScope
+import org.oppia.app.model.Voiceover
+import org.oppia.domain.audio.AudioPlayerController
+import org.oppia.domain.audio.AudioPlayerController.PlayProgress
+import org.oppia.domain.audio.AudioPlayerController.PlayStatus
+import org.oppia.util.data.AsyncResult
 import javax.inject.Inject
 
 /** [ViewModel] for audio-player state. */
 @FragmentScope
 class AudioViewModel @Inject constructor(
+  private val audioPlayerController: AudioPlayerController
 ) : ViewModel() {
-  val currentLanguageCode = ObservableField<String>("en")
 
+  private lateinit var explorationId: String
+  private var voiceoverMap = mapOf<String, Voiceover>()
+
+  /** Mirrors PlayStatus in AudioPlayerController except adds LOADING state */
+  enum class AudioPlayStatus {
+    LOADING,
+    PREPARED,
+    PLAYING,
+    PAUSED,
+    COMPLETED
+  }
+
+  val currentLanguageCode = ObservableField<String>()
+
+  val durationLiveData: LiveData<Int> by lazy {
+    processDurationLiveData()
+  }
+  val positionLiveData: LiveData<Int> by lazy {
+    processPositionLiveData()
+  }
+  val playStatusLiveData: LiveData<AudioPlayStatus> by lazy {
+    processPlayStatusLiveData()
+  }
+
+  fun setVoiceoverMappings(map : Map<String, Voiceover>) {
+    voiceoverMap = map
+  }
+
+  fun setExplorationId(id: String) {
+    explorationId = id
+  }
+
+  /** Sets language code for data binding and changes data source to correct audio */
   fun setAudioLanguageCode(languageCode: String) {
     currentLanguageCode.set(languageCode)
+    audioPlayerController.changeDataSource(voiceOverToUri(voiceoverMap[languageCode]))
+  }
+
+  /** Plays or pauses AudioController depending on passed in state */
+  fun handlePlayPause(type: AudioPlayStatus?) {
```

Comment:
I wonder if we can make the name more clear so that it's clearer at callsites what calling this method actually does. Though when reading this, I'm wondering if we actually want to call it in the places that we are. Perhaps we really want two methods: one that handles toggling play/pause in the UI, and another that properly auto starts/stops it based on lifecycles. That will result in more predictability, I think.